cloudcix_primitives/network_secondary_lxd.py

```diff
@@ -0,0 +1,249 @@
+
+__all__ = [
+    'build',
+    'read',
+    'scrub',
+]
+
+def build(
+    endpoint_url: str,
+    project: str,
+    container_name: str,
+    vlan_id: str,
+    device_name: str,
+    mac_address: str = None,
+    verify_lxd_certs: bool = True,
+) -> Tuple[bool, str]:
+    """
+    Attach a secondary network interface to an LXD container.
+    """
+    # Define the messages
+    messages = {
+        1000: f'Successfully attached network interface to container {container_name} on {endpoint_url}',
+        3021: f'Failed to connect to {endpoint_url} for network or container operations',
+        3022: f'Failed to retrieve networks from {endpoint_url}. Payload exited with status ',
+        3023: f'Failed to retrieve container {container_name}. Payload exited with status ',
+        3024: f'Network device {device_name} already exists on container {container_name}.',
+        3025: f'Network br{vlan_id} does not exist on {endpoint_url}.',
+        3026: f'Failed to create network device: ',
+    }
+
+    def run_host(endpoint_url, prefix, successful_payloads):
+        rcc = LXDCommsWrapper(comms_lxd, endpoint_url, verify_lxd_certs, project)
+        fmt = HostErrorFormatter(
+            endpoint_url,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+
+        network_interface_name = f'br{vlan_id}'
+
+        # Check if the network exists
+        ret = rcc.run(cli='networks.all')
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+2}: {messages[prefix+2]}"), fmt.successful_payloads
+
+        networks = ret['payload_message']
+        fmt.add_successful('networks.all', ret)
+
+        # Check if our network exists
+        network_exists = False
+        for network in networks:
+            if network.name == network_interface_name:
+                network_exists = True
+
+        if not network_exists:
+            return False, f"{prefix+5}: {messages[prefix+5]}", fmt.successful_payloads
+
+        # Get the container
+        ret = rcc.run(cli='containers.get', name=container_name)
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+3}: {messages[prefix+3]}"), fmt.successful_payloads
+
+        instance = ret['payload_message']
+        fmt.add_successful('containers.get', ret)
+
+        # Check if the device already exists
+        devices = instance.devices
+        if device_name in devices:
+            return False, f"{prefix+4}: {messages[prefix+4]}", fmt.successful_payloads
```

Comment:
If the `device_name` already exists, this is okay, we can return True as it is the expected result from a build